src/clients/feed/feed.ts

```diff
@@ -161,6 +164,59 @@ class Feed {
     return this.makeStatusUpdate(itemOrItems, "seen");
   }
 
+  async markAllAsSeen() {
+    // To mark all of the messages as seen we:
+    // 1. Optimistically update *everything* we have in the store
+    // 2. We decrement the `unseen_count` to zero optimistically
+    // 3. We issue the API call to the endpoint
+    //
+    // Note: there is the potential for a race condition here because the bulk
+    // update is an async method, so if a new message comes in during this window before
+    // the update has been processed we'll effectively reset the `unseen_count` to be what it was.
+    //
+    // Note: here we optimistically handle the case whereby the feed is scoped to show only `unseen`
+    // items by removing everything from view.
+    const { getState, setState } = this.store;
+    const { metadata, items } = getState();
+
+    const isViewingOnlyUnseen = this.defaultOptions.status === "unseen";
+
+    // If we're looking at the unseen view, then we want to remove all of the items optimistically
+    // from the store given that nothing should be visible. We do this by resetting the store state
+    // and setting the current metadata counts to 0
+    if (isViewingOnlyUnseen) {
+      setState((store) =>
+        store.resetStore({
+          ...metadata,
+          total_count: 0,
+          unseen_count: 0,
+        }),
+      );
+    } else {
+      // Otherwise we want to update the metadata and mark all of the items in the store as seen
+      setState((store) => store.setMetadata({ ...metadata, unseen_count: 0 }));
+
+      const attrs = { seen_at: new Date().toISOString() };
```

Comment:
Because in the unseen view case we're **wiping** all items in the store (e.g. removing everything) as they've now been seen and therefore should no longer be in the store. That's what the `resetStore` method does on line 189 here